tests/test_cli.py

```diff
@@ -25,3 +28,39 @@ def test_namespace(namespace, expected, runner):
     assert result.exit_code == 0
 
     assert expected in result.output
+
+
+@pytest.mark.skip  # skipped until we solve all conflicts
+def test_options_structure(runner):
+    result = runner.invoke(["--options-structure"])
+    assert result.exit_code == 0
+
+    options_json = json.loads(result.output)
+    assert find_conflicts_in_options_dict("snow", options_json) is None
+
+
+def find_conflicts_in_options_dict(path: str, options_dict: Dict[str, Any]):
+
+    keys = list(options_dict.keys())
+    duplicates = {}
+    if "options" in keys:
+        if opts_duplicates := check_options_for_duplicates(options_dict["options"]):
+            duplicates[path] = opts_duplicates
+        keys.remove("options")
+
+    for key in keys:
+        if key_duplicates := find_conflicts_in_options_dict(key, options_dict[key]):
+            duplicates[key] = key_duplicates
```

Comment:
```suggestion
        if key_duplicates := find_conflicts_in_options_dict(f"{path} {key}", options_dict[key]):
            duplicates.update(key_duplicates)
```

I think it produces more readable output.

<details>
<summary>Your output (click to expand)</summary>

```
{
  'connection': {
    'add': {
      'add': {
        '-h'
      }
    }
  },
  'snowpark': {
    'function': {
      'create': {
        'create': {
          '-h'
        }
      },
      'update': {
        'update': {
          '-h',
          '-a'
        }
      }
    },
    'jobs': {
      'create': {
        'create': {
          '-c'
        }
      },
      'logs': {
        'logs': {
          '-c'
        }
      }
    },
    'procedure': {
      'create': {
        'create': {
          '-h'
        }
      },
      'describe': {
        'describe': {
          '-p'
        }
      },
      'drop': {
        'drop': {
          '-p'
        }
      },
      'execute': {
        'execute': {
          '-p'
        }
      },
      'update': {
        'update': {
          '-h'
        }
      }
    },
    'services': {
      'create': {
        'create': {
          '-p'
        }
      }
    }
  }
}
```

</details>

<details>
<summary>My output (click to expand)</summary>

```
{
  'snow connection add': {
    '-h'
  },
  'snow snowpark function create': {
    '-h'
  },
  'snow snowpark function update': {
    '-a',
    '-h'
  },
  'snow snowpark jobs create': {
    '-c'
  },
  'snow snowpark jobs logs': {
    '-c'
  },
  'snow snowpark procedure create': {
    '-h'
  },
  'snow snowpark procedure describe': {
    '-p'
  },
  'snow snowpark procedure drop': {
    '-p'
  },
  'snow snowpark procedure execute': {
    '-p'
  },
  'snow snowpark procedure update': {
    '-h'
  },
  'snow snowpark services create': {
    '-p'
  }
}
```

</details>

Reply:
Good point. Done